core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -711,14 +731,18 @@ public FolderId getFolderIdByFolderName(ExchangeService exchangeService, String
 
 	@Override
 	public void createFolder(String folderName) throws FileSystemException {
-		ExchangeService exchangeService = getConnection();
+		String folderNameToUse = getFolderNameToUse(folderName);
+		String mailbox = getMailboxToUse(folderName);
+		ExchangeService exchangeService = getConnection(mailbox);
+
 		try {
 			Folder folder = new Folder(exchangeService);
-			folder.setDisplayName(folderName);
-			folder.save(new FolderId(basefolderId.getUniqueId()));
+			folder.setDisplayName(folderNameToUse);
+			folder.save(cache.getBaseFolderId(mailbox));
+			cache.registerFolder(mailbox, folder);
```

Comment:
Je stopt hier de folder in de cache, om hem later weer te kunnen gebruiken zonder dat je een exchangeService hoeft te hebben. Daarmee speel je vals, want dit folder-object is onder water toch nog gekoppeld aan de service die hem gecreerd heeft. Als je er nog wat mee doet, dan maak je gebruik van een service(connectie) die je al weer teruggegeven hebt aan de pool.
Als dat zou kunnen, dan was dat hele mechanisme met connecties en connectiepools in de first place niet nodig, dan kon je gewoon één centrale service gebruiken.
Met andere woorden, ik denk dat je threading problemen gaat krijgen.

Reply:
Het threading issue speelt enkel wanneer je meerdere threads gebruik laat maken van het zelfde service object.

Als het folder object daadwerkelijk nog gekoppeld is met de service die het folder object gemaakt heeft, dan zou ik verwachten dat de testen ook falen.

Daar wordt er immers eerst een folder aangemaakt door een service verkregen uit de pool, deze wordt terug gegeven en vervolgens wordt er weer een service uit de pool opgevraagd door removeFolder.

Of spelen binnen de integratie testen de threading issues mogelijk niet omdat er altijd slechts een service uit de pool wordt gebruikt? Is dat per definitie dan altijd dezelfde?

Reply:
De integratietesten testen inderdaad niet de ThreadSafe-heid

Reply:
Overal waar cache.getFolder wordt aangeroepen, wordt vervolgens .getId op het folder object aangeroepen (behalve in removeFolder).

Het FolderId object houd de service niet intern nog vast en kan dus wel hergebruikt worden. In de cache dient het Folder object vervangen te worden met FolderId.

In de removeFolder methode mag het folderId wel uit cache gelezen worden maar moet er met behulp van een service alsnog het daadwerkelijke Folder object opgehaald worden.

Reply:
Dat lijkt me een goede aanpak